test/test_linalg.py

```diff
@@ -1327,29 +1328,49 @@ def gen_error_message(input_size, ord, keepdim, dim=None):
             raise RuntimeError(f'dtype not supported in this test: {dtype}')
 
         vector_ords = [None, 0, 1, 2, 3, inf, -1, -2, -3, -inf]
-        matrix_ords = [None, 'fro', 1, inf, -1, -inf]
-
-        # TODO: Fix autograd for matrix orders 'nuc', 2, and -2 by adding complex
-        #       support to svd's backward method. Once this is done, these ords
-        #       should be added to `matrix_ords` above
-        # Update: svd's backward now works with https://github.com/pytorch/pytorch/pull/47761
-        # However run_test_case doesn't work for 'matrix_ords_unsupported' cases
-        # because singular values of 'x' and 'x_real' can be different and so is their norms based on singular values
-        matrix_ords_unsupported = ['nuc', 2, -2]
+        matrix_ords = [None, 'fro', 'nuc', 1, 2, inf, -1, -2, -inf]
 
         def run_test_case(x, ord, keepdim):
+            msg = gen_error_message(x.size(), ord, keepdim)
+
             res = torch.linalg.norm(x, ord, keepdim=keepdim)
             res.backward()
 
-            x_real = x.clone().detach().abs().requires_grad_(True)
-            res_real = torch.linalg.norm(x_real, ord, keepdim=keepdim)
-            res_real.backward()
+            is_nuclear = isinstance(ord, str) and ord == 'nuc'
+            is_matrix_2 = isinstance(ord, Number) and abs(ord) == 2
 
-            msg = gen_error_message(x.size(), ord, keepdim)
+            # Matrix norms order +/-2 and Nuclear can be compared with SVD
```

Comment:
@kurtamohler why don't we just use `gradcheck` here? We should also consider adding an `OpInfo` for `torch.linalg.norm` I think 

cc. @mruberry

Reply:
For some reason I was under the impression that `gradcheck` only works for `torch.double`, but it does indeed work with `torch.cdouble` as well. However, it does not work for `torch.float` or `torch.cfloat`: https://github.com/pytorch/pytorch/blob/master/torch/autograd/gradcheck.py#L309

I assume it's acceptable not to include autograd tests for `torch.float` and `torch.cfloat`, so I'll update to use `gradcheck`. In that case, I can actually just remove `test_norm_complex_autograd` and add `torch.cdouble` to `test_autograd_and_jit`, which already uses `gradcheck`.

Reply:
You're correct, @kurtamohler, that it's only intended to support double and cdouble, and we typically only test grad and gradgrad in those dtypes.

I think it is time to implement an OpInfo for torch.linalg.norm. That will automatically generate grad and gradgrad tests for all of its sample inputs. Ping me if you have questions about how to do this.

Reply:
But yes, `OpInfo` seems like a good idea. I'll look into that now